airbyte-webapp/src/components/connectorBuilder/useManifestToBuilderForm.ts

```diff
@@ -168,100 +169,102 @@ const manifestStreamToBuilder = (
+      throw new ManifestCompatibilityError(streamName, "SubstreamPartitionRouter has more than one parent stream");
     }
-    const parentStreamConfig = manifestSubstreamSlicer.parent_stream_configs[0];
+    const parentStreamConfig = manifestSubstreamPartitionRouter.parent_stream_configs[0];
 
     const matchingStreamIndex = serializedStreamToIndex[JSON.stringify(parentStreamConfig.stream)];
     if (matchingStreamIndex === undefined) {
       throw new ManifestCompatibilityError(
         streamName,
-        "SubstreamSlicer's parent stream doesn't match any other stream"
+        "SubstreamPartitionRouter's parent stream doesn't match any other stream"
       );
     }
 
-    return {
-      type: "SubstreamSlicer",
-      parent_key: parentStreamConfig.parent_key,
-      stream_slice_field: parentStreamConfig.stream_slice_field,
-      parentStreamReference: matchingStreamIndex.toString(),
-    };
+    return [
+      {
+        type: "SubstreamPartitionRouter",
+        parent_key: parentStreamConfig.parent_key,
+        partition_field: parentStreamConfig.partition_field,
+        parentStreamReference: matchingStreamIndex.toString(),
+      },
+    ];
   }
 
-  throw new ManifestCompatibilityError(streamName, "stream_slicer type is unsupported");
+  throw new ManifestCompatibilityError(streamName, "partition_router type is unsupported");
 }
 
 function manifestPrimaryKeyToBuilder(manifestStream: DeclarativeStream): BuilderStream["primaryKey"] {
-  if (!isEqual(manifestStream.primary_key, manifestStream.retriever.primary_key)) {
+  if (!isEqual(manifestStream.primary_key, manifestStream.primary_key)) {
     throw new ManifestCompatibilityError(
       manifestStream.name,
       "primary_key is not consistent across stream and retriever levels"
     );
   }
   if (manifestStream.primary_key === undefined) {
     return [];
-  } else if (Array.isArray(manifestStream.retriever.primary_key)) {
-    if (manifestStream.retriever.primary_key.length > 0 && Array.isArray(manifestStream.retriever.primary_key[0])) {
+  } else if (Array.isArray(manifestStream.primary_key)) {
+    if (manifestStream.primary_key.length > 0 && Array.isArray(manifestStream.primary_key[0])) {
       throw new ManifestCompatibilityError(manifestStream.name, "primary_key contains nested arrays");
     } else {
-      return manifestStream.retriever.primary_key as string[];
+      return manifestStream.primary_key as string[];
     }
   } else {
-    return [manifestStream.retriever.primary_key];
+    return [manifestStream.primary_key];
   }
 }
 
+function manifestIncrementalSyncToBuilder(
+  manifestIncrementalSync: DeclarativeStreamIncrementalSync | undefined,
+  streamName?: string
+): DatetimeBasedCursor | undefined {
+  if (!manifestIncrementalSync) {
+    return undefined;
+  }
+  if (manifestIncrementalSync.type === "CustomIncrementalSync") {
+    throw new ManifestCompatibilityError(streamName, "incremental sync uses a custom implementation");
```

Comment:
Just for my understanding - we throw here because the custom implementation could not be represented in the builder form UI?

Reply:
exactly, custom implementations require python files we can't handle in the UI, so we prevent conversion and the user needs to stick in the yaml editor (there are a bunch of these cases).